Source/WebCore/platform/graphics/TiledBacking.h

```diff
@@ -33,6 +33,10 @@
 #include <wtf/MonotonicTime.h>
 #include <wtf/WeakPtr.h>
 
+#if ENABLE(RE_DYNAMIC_CONTENT_SCALING)
+#include "DynamicContentScalingDisplayList.h"
```

Comment:
Nit: you can forward declare `class DynamicContentScalingDisplayList;` since `std::optional` doesn't need the full type definition for the return type.

Reply:
Same comment about some of the other header files

Reply:
No, I don't think that's correct.
std::optional doesn't play a role here.

These behave the same way:
```
std::optional<DynamicContentScalingDisplayList> a();
DynamicContentScalingDisplayList b();
```

If your rule was correct rule, mostly nothing would need to be included.

The fact is that the compiler needs the type to be complete when the function is called, in both cases.

The convention is that it's poor form to expect the caller to understand that they would need to include "DynamicContentScalingDisplayList" before calling.

Thus the header declaring a function returning DynamicContentScalingDisplayList by value should include the header. Returning it through std::optional is still returning it by value.